packages/connection/index.js

```diff
@@ -106,49 +107,50 @@ class Connection extends EventEmitter {
     return this.jid
   }
 
-  _online() {
-    this.emit('online', this.jid)
-  }
-
-  _authenticated() {
-    this.emit('authenticated')
+  _status(status, ...args) {
+    this.status = status
+    this.emit('status', status, ...args)
+    this.emit(status, ...args)
```

Comment:
Yet you convinced me. :-)

The promise I want to get resolved when online (or rejected if the "getting online" process failed for some reason) is the promise returned from `.start()`.

So, we should have:
1. `client.on('adjective', listener)`
2. `client.verb(opts).then(...)`